primer/react
# feat(DataTable): add support for overflow behavior

Closes https://github.com/github/primer/issues/1835

Add support for overflow to `Table`. This will create a `role="region"` with a `tabindex` set so that when the table is scrollable someone may interact with it (reference axe rule: https://dequeuniversity.com/rules/axe/4.0/scrollable-region-focusable)

This also adds in an e2e test for this behavior along with screenshots for the various states


### Testing & Reviewing

- Visit the `With overflow` story in storybook
- Verify that the `<table>` element is wrapped in a `div` with `role="region"`
  - This region should be focusable
- Visit another story in the storybook
- Resize the viewport until overflow occurs
- Verify that the `<table>` element is wrapped in a `div` with `role="region"`, as above
- Resize the viewport until overflow no longer occurs
- Verify that the `<table>` element is no longer wrapped in a `div`
